# Homeworl 4 Assignment 3

This notebook is for Assignment 3 of Homework 4 for PUI2018. This notebook uses data from a text file for the (fictitious) 
bus line X8. And compare the new times to the previous mean and SD times for the bus which took 36 +- 6 minutes 
(mean and standard deviation parameters of the population). 

For this homeworkassignment, I worked with Sam Falk and Qin Yu

## Null Hypothesis
$H_0$ : The mean commute time after implementation of the new route for bus line X8 is the same or longer during peak hours as compared to prior mean of 36 minutes with a significance level p = 0.05.  

## Alternative Hypothesis

$H_A$ : The mean commute time after implementation of the new route for bus line X8 is shorter during peak hours as compared to prior mean of 36 minutes with a significance level p = 0.05.


$H_0$ : $P_1$ - $P_0$ => 0

$H_1$ : $P_1$ - $P_0$ < 0

α = 0.05

2 or more sigmas away from old mean then reject NULL using Z-test

In [1]:
from __future__ import print_function #for compatibility beteen python2 and python3

In [7]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > x8_bus.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0  11310      0 --:--:-- --:--:-- --:--:-- 11336


In [8]:
# check to see if there is a PUIDATA env variable
import os

print("PUIdata", os.getenv("PUIDATA"))

PUIdata /nfshome/yg833/PUI2018/PUIdata


In [9]:
os.environ["PUIDATA"] = "%s/PUI2018/PUIdata"%os.getenv("HOME") ## force the PUIdata variable

print("PUIdata", os.getenv("PUIDATA"))

PUIdata /nfshome/yg833/PUI2018/PUIdata


In [10]:
# move the data
!mv x8_bus.txt $PUIDATA

In [13]:
#read in the data with pandas
import pandas as pd
import numpy as np
bus_df = pd.read_table("%s/x8_bus.txt"%os.getenv("PUIDATA"), delim_whitespace=True, names=("t"))

In [18]:
# calculate the new mean

sample_mean = bus_df.mean()

sample_mean

t    34.466162
dtype: float64

In [19]:
# calcuate the SD

sample_sd = bus_df.std()

sample_sd

t    7.13728
dtype: float64

In [21]:
# get the sample size

sample_N = len(bus_df)

sample_N

100

In [22]:
# population mean and SD
pop_mean = 36

pop_sd = 6

In [24]:
# calculate Z statistic

z_test = (pop_mean - sample_mean)/ (pop_sd/((sample_N)**(1/2)))

z_test


t    2.556397
dtype: float64

We can reject the null as the Z-score of 2.56 is more than 2 sigmas away from the old mean. 